### DISCUSSION TITLE
Write to StandardInput in ToolTask

### DISCUSSION THREAD

**Discussion by magol** [dotnet/msbuild#7322] [Q&A] [answered]
I want to pipe data to a command line tool. I wonder if it is possible to use [Process.StandardInput](https://msdn.microsoft.com/en-us/library/system.diagnostics.process.standardinput) to write to do that? I see that StandardInput is used in the [source code](https://github.com/Microsoft/msbuild/blob/6851538897f5d7b08024a6d8435bc44be5869e53/src/Utilities/ToolTask.cs#L868), but I don't understand how I can use this in my own code.

**Comment by rainersigwald** [comment, marked as answer]
It doesn't look like ToolTask provides hooks for that, so you'd have to either implement it yourself (in a different `ITask` implementation) or override `ToolTask.ExecuteTool` with the behavior you want.
